### PR TITLE
[CR] Anvil tongs and linting changes

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
Attempted to generify anvil and tongs by adding ANVIL and grip tool qualities.
Small and medium boulders can now be used as anvils in some recipes, copper items and some HAMMER recipes now require ANVIL, and the Medieval and Historic Content mod now has a bronze anvil that can be made with bronze and used to make bronze tools.
Pliers, multitools, tongs, and integrated toolsets have gained GRIP tool quality, and a handful of recipes have been updated to use it.

The anvil now uses the deployable items code instead of the construction menu.

I'm going to end this pr with a lotta linting.

### CODE REVIEW

**Comment by Cyrano7** [comment]
> I'm going to end this pr with a lotta linting.

Mixing linting with content changes in a 3k+ PR is not really reviewable. At least not for me.

**Comment by BorkBorkGoesTheCode** [comment]
>Mixing linting with content changes in a 3k+ PR is not really reviewable. At least not for me.

The linting is separate from the code changes, so that isn't an issue unless your UI doesn't show the names of changed files.

**Comment by smolbird** [comment]
Isn't there also a mod that does that with boulders?

**Comment by BorkBorkGoesTheCode** [comment]
>Isn't there also a mod that does that with boulders?

Mining Mod has a recipe which effectively converts boulders into steel anvils, meaning that the regular steel anvil is only useful in cities away from boulders. This PR creates lower tiers of anvil using tool qualities, and makes anvils easier to mod with.

File: data/json/furniture.json [hunk 1]
```diff
@@ -2070,12 +2070,9 @@
...
         "required_str": 16,
         "crafting_pseudo_item": "anvil",
-        "deconstruct": {
-            "items": [
-                { "item": "anvil", "count": 1 }
-            ]
-        },
-        "flags": ["TRANSPARENT", "NOITEM", "EASY_DECONSTRUCT"],
+        "flags": ["PLACE_ITEM", "TRANSPARENT", "NOITEM", "EASY_DECONSTRUCT"],
+        "deployed_item": "brazier",
```
Comment (smolbird) [inline data/json/furniture.json]: Turns into a brazier when taken down?

**Comment by smolbird** [comment]
Ah that's it, yeah. Will have to see what they do with it when this PR is merged.

Oh wait, actually @DangerNoodle, @chaosvolt, @pisskop, and @Noctifer-de-Mortem will all want to know about this change, if tongs and anvils are becoming a quality thing.

**Comment by BorkBorkGoesTheCode** [comment]
I'm dropping this for now.

File: data/json/furniture.json (hunk 1 above)
Comment (BorkBorkGoesTheCode) [reply data/json/furniture.json]: Fixed that now.

**Comment by BorkBorkGoesTheCode** [comment]
A cleaner PR without deployable furniture is here: https://github.com/CleverRaven/Cataclysm-DDA/pull/23637

**Pull request closed without merging**